clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -681,6 +669,11 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
      * errors, this method will throw the last received exception immediately and the transaction will not be committed.
      * So all {@link #send(ProducerRecord)} calls in a transaction must succeed in order for this method to succeed.
      *
+     * Note that this method will raise {@link TimeoutException} if the transaction cannot be committed before expiration
+     * of {@code max.block.ms}. Additionally, it will raise {@link InterruptException} if interrupted.
+     * It is safe to retry in either case, but once the transaction has been successfully committed,
```

Comment:
Minor suggestion:

> It is safe to retry in either case, but it is not possible to attempt a different operation (such as `abortTransaction`) since the commit may already be in the progress of completing. If not retrying, the only option is to close the producer.